src/scalar_field.nr

```diff
@@ -153,3 +145,19 @@ impl<let N: u32> ScalarField<N> {
         self.base4_slices[idx]
     }
 }
+
+#[test]
+fn test_wnaf() {
+    unsafe {
+        let result: Field = 0x123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef0;
+        let (t0, t1) = get_wnaf_slices::<64>(result);
+        let expected = from_wnaf_slices(t0, t1);
+        assert(result == expected);
+    }
+}
+
+#[test]
+fn test_regression_wnaf() {
+    let a = 0x7b;
+    let _: ScalarField<64> = ScalarField::<64>::from(a);
```

Comment:
I know this would previously trigger an assertion failure but, for completeness, we could change this to be:
```suggestion
    let s: ScalarField<64> = ScalarField::<64>::from(a);
    let b: Field = s.into();
    assert_eq(a, b);
```